File: SmallestMultiple.py
#!/usr/bin/python3

def main():
    counter = getMultiplesOfPrime(20)
    filterDevision = getMultiplesOfPrime(20)
    while True:
        if counter % filterDevision != 0:
            counter += 1
            continue
        print(counter)
        for d in range(1,21):
            if counter % d != 0:
                counter += 1
                break
        else:
            print("The answer is", counter)
            break

def isPrime(number):
    if number == 1:
        return False
    for n in range(2, number):
        if number % n == 0:
            return False
    else:
        return True
       
def PrimeGenerator():
    counter = 1
    while True:
        if isPrime(counter): yield counter
        counter += 1
        
def getMultiplesOfPrime(maxDevision):
    number = 1
    for prime in PrimeGenerator():
        if prime <= maxDevision:
            number *= prime
        else:
            return number
        
if __name__ == "__main__":main()

File: SmallestMultiple.cpp
//Created on: Dec 17, 2020 by Alireza
#include<iostream>
#include<math.h>
using namespace std;

bool isPrime(int number);
int Prime(int index);

int main(int argc, char ** argv){
	int limit = 20;
	long long unsigned int answer = 1;
	for(int i = 0; i <= limit; ++i){
		int power = 0;
		int prime = Prime(i);
		if(prime <= limit){
			power = floor((log(limit)) / (log(prime)));
			answer *= (int)(pow(prime, power));
		}
	}
	cout << "The answer is " << answer << endl;
	return 0;
}

bool isPrime(int number){
	if(number <= 1) return false;
	for(int i = 2; i < ((int)(sqrt(number))) + 1; ++i){
		if(number % i == 0) return false;
	}
	return true;
}

int Prime(int index){
	int counter = 0;
	int number = 2;
	while(true){
		if(isPrime(number)){
			if(counter == index){
				return number;
			}
			else{
				counter++;
			}
		}
		number++;
	}
}
